pkg/envoy/mock_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017-2019 Authors of Cilium
```

Comment:
This file doesn't have any tests in it. Is there a reason it is named "mock_test"? Could it just be "mock.go"?

Reply:
We could also introduce a `mock` subpackage inside the `envoy` package, that way we don't pollute the `envoy` package namespace and reduce the burden of importing it.

Reply:
Ah, didn't see the conversation about this earlier. Please ignore.

Reply:
@christarazi The mock is used from another *_test.go file, and need not be linked into the release code. My understanding is that Go does not automatically exclude dead code.